cpp/modules/deck.gl/core/src/lib/deck.cc

```diff
@@ -72,6 +72,10 @@ Deck::Deck(std::shared_ptr<Deck::Props> props)
       lumagl::utils::createBuffer(this->animationLoop->device(), sizeof(ViewportUniforms), wgpu::BufferUsage::Uniform);
 }
 
+auto Deck::make(probegl::Error* error, std::shared_ptr<Deck::Props> props) -> std::optional<Deck> {
```

Comment:
Naming: `make_shared`, and return shared pointer? S

lightly less general, slightly safer, more easy for under to understand... 

The rest of the API already prescribes shared_ptr so I don't think we are assuming to much>